rllib/evaluation/worker_set.py

```diff
@@ -408,12 +413,8 @@ def sync_weights(
         weights = None
         if self.num_remote_workers() or from_worker is not None:
             weights = (from_worker or self.local_worker()).get_weights(policies)
-            # Put weights only once into object store and use same object
-            # ref to synch to all workers.
-            weights_ref = ray.put(weights)
-
             def set_weight(w):
-                w.set_weights(ray.get(weights_ref), global_vars)
+                w.set_weights(weights, global_vars)
```

Comment:
would `from_worker.get_weights(policies)` return an object ref, or the actual weights?

If it's an object ref, would `set_weight` work for the case where from_worker is a remote worker? If this is the case do we test this behavior in a unit test?

Reply:
from_worker has to be a local RolloutWorker. so ``weights`` here must be raw weights.

the reason we have from_worker is that oftentimes, evaluation_worker_set doesn't have a local worker to sync from.
you need to sync weights from rollout_workers.local_worker() to evaluation_workers.remote_workers().

also, the reason I got rid of ray.get/put here is that, when testing everything, I noticed some slight improvements if we don't force ray.put on every single weights dict. seems like Ray core may optimize things and say if all the remote workers are on the same instance, skip serialization and simply copy over the data. need to confirm this though.